### PR TITLE
Move the roles to security as an Enum like class

### PR DESCRIPTION
**Pull request by jon-betts**
A real enum doesn't work here for some reason. Again this might be
something we can fix in future.

### CODE REVIEW

File: tests/h/traversal/group_test.py [hunk 1]
```diff
@@ -17,22 +15,6 @@ def test_it_with_a_group(self, factories, ACL):
...
-        set_permissions("acct:adminuser@foo", principals=[principal])
-
-        context = GroupContext(group=None)
-
-        assert context.group is None
-        assert (
-            bool(pyramid_request.has_permission(Permission.Group.UPSERT, context))
-            == has_upsert
-        )
-
```
Comment (jon-betts) [inline tests/h/traversal/group_test.py:35]: I noticed that the role was being used here where it really shouldn't be. There was some old left over tests from before the ACL move.

File: tests/h/security/test_acl.py [hunk 2]
```diff
@@ -75,7 +75,8 @@ def test_it(self, user_permits, user, principal_template, permission, is_permitt
     def test_a_user_is_not_required_for_create(self, permits):
         acl_object = ObjectWithACL(ACL.for_user(user=None))
 
-        permits(acl_object, [role.AuthClient], Permission.User.CREATE)
+        assert permits(acl_object, [Role.AUTH_CLIENT], Permission.User.CREATE)
+        assert not permits(acl_object, [Role.AUTH_CLIENT], Permission.User.READ)
```
Comment (jon-betts) [inline tests/h/security/test_acl.py:79]: I spotted we weren't actually making any assertions here. Also, due to our generator approach and the fact that Pyramid stops as soon as it gets the permission it needs we weren't getting coverage of one of the lines.

The negative permission check forces Pyramid to exhaust the generator.

File: tests/h/security/test_acl.py [hunk 3]
```diff
@@ -105,28 +106,31 @@ def test_it(self, permits, client_authority, principal, is_permitted):
             == is_permitted
         )
 
+    def test_it_without_client_authority(self):
+        assert not list(ACL.for_bulk_api(client_authority=None))
```
Comment (jon-betts) [inline tests/h/security/test_acl.py:110]: Some missing coverage here too.

File: tests/h/security/test_acl.py [hunk 4]
```diff
@@ -335,6 +339,15 @@ def test_it_mirrors_permissions_from_the_group_for_shared_annotations(
...
         for_group.assert_called_with(group)
 
+    def test_it_mirrors_group_permissions_for_shared_deleted_annotations(
+        self, annotation, group, for_group, anno_permits
+    ):
+        annotation.shared = True
+        annotation.deleted = True
+        for_group.return_value = [(security.Allow, "principal", Permission.Group.READ)]
+
+        anno_permits(["principal"], Permission.Annotation.READ_REALTIME_UPDATES)
```
Comment (jon-betts) [inline tests/h/security/test_acl.py:349]: More missing coverage...

**Review by marcospri — APPROVED** [review]

**Pull request merged**